packages/db/tests/models/test-granules-executions-model.js

```diff
@@ -0,0 +1,194 @@
+const cryptoRandomString = require('crypto-random-string');
+
+const {
+  CollectionPgModel,
+  destroyLocalTestDb,
+  ExecutionPgModel,
+  fakeCollectionRecordFactory,
+  fakeExecutionRecordFactory,
+  fakeGranuleRecordFactory,
+  generateLocalTestDb,
+  GranulePgModel,
+  GranulesExecutionsPgModel,
+  tableNames,
+} = require('../../dist');
+
+const { migrationDir } = require('../../../../lambdas/db-migration');
+
+const testDbName = `granule_${cryptoRandomString({ length: 10 })}`;
+
+test.before(async (t) => {
+  const { knexAdmin, knex } = await generateLocalTestDb(
+    testDbName,
+    migrationDir
+  );
+  t.context.knexAdmin = knexAdmin;
+  t.context.knex = knex;
+
+  t.context.granulePgModel = new GranulePgModel();
+  t.context.granulesExecutionsPgModel = new GranulesExecutionsPgModel();
+
+  const collectionPgModel = new CollectionPgModel();
+  [t.context.collectionCumulusId] = await collectionPgModel.create(
+    t.context.knex,
+    fakeCollectionRecordFactory()
+  );
+
+  t.context.executionPgModel = new ExecutionPgModel();
+});
+
+test.beforeEach(async (t) => {
+  const [executionCumulusId] = await t.context.executionPgModel.create(
+    t.context.knex,
+    fakeExecutionRecordFactory()
+  );
+  t.context.executionCumulusId = executionCumulusId;
+  const [granuleCumulusId] = await t.context.granulePgModel.create(
+    t.context.knex,
+    fakeGranuleRecordFactory({
+      collection_cumulus_id: t.context.collectionCumulusId,
+    })
+  );
+  t.context.granuleCumulusId = granuleCumulusId;
+  t.context.joinRecord = {
+    execution_cumulus_id: executionCumulusId,
+    granule_cumulus_id: Number(granuleCumulusId),
+  };
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex(tableNames.granulesExecutions).where(t.context.joinRecord).del();
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test('GranulesExecutionsPgModel.create() creates a new granule/execution join record', async (t) => {
+  const {
+    knex,
+    granulesExecutionsPgModel,
+    joinRecord,
+  } = t.context;
+
+  t.plan(1);
+
+  await knex.transaction(async (trx) => {
+    await granulesExecutionsPgModel.create(trx, joinRecord);
+    const records = await trx(tableNames.granulesExecutions).where(joinRecord);
+    t.is(
+      records.length,
+      1
+    );
+  });
+});
+
+test('GranulesExecutionsPgModel.exists() correctly returns true', async (t) => {
+  const {
+    knex,
+    granulesExecutionsPgModel,
+    joinRecord,
+  } = t.context;
+
+  t.plan(1);
+
+  await knex.transaction(async (trx) => {
+    await granulesExecutionsPgModel.create(trx, joinRecord);
+    t.true(
+      await granulesExecutionsPgModel.exists(trx, joinRecord)
+    );
+  });
+});
+
+test('GranulesExecutionsPgModel.exists() correctly returns false', async (t) => {
+  const {
+    knex,
+    granulesExecutionsPgModel,
+  } = t.context;
+
+  t.plan(1);
+
+  await knex.transaction(async (trx) => {
```

Comment:
Should this test be .serial given it's *possible* a 3 digit random crypto will result in 5:5 for the IDs in another test?

Reply:
Yeah I couldn't think of how to make it safe to pick a set of IDs that shouldn't exist. Good call.

Reply:
67506a7de1d571cc58ecceb3f662b489a0e1198d